File: java-core-plugins/src/main/java/org/unlaxer/tramli/plugins/diagram/DiagramGenerationPlugin.java
package org.unlaxer.tramli.plugins.diagram;

import org.unlaxer.tramli.FlowDefinition;
import org.unlaxer.tramli.FlowState;
import org.unlaxer.tramli.plugins.api.GenerationPlugin;
import org.unlaxer.tramli.plugins.api.PluginDescriptor;

public final class DiagramGenerationPlugin<S extends Enum<S> & FlowState> implements GenerationPlugin<FlowDefinition<S>, DiagramBundle> {
    private final DiagramPlugin delegate = new DiagramPlugin();

    @Override
    public PluginDescriptor descriptor() {
        return new PluginDescriptor("diagram", "Diagram Generator", "Generates Mermaid and data-flow bundles from a flow definition.");
    }

    @Override
    public DiagramBundle generate(FlowDefinition<S> input) {
        return delegate.generate(input);
    }
}

File: lang/ts-plugins/src/diagram/diagram-generation-plugin.ts
import type { FlowDefinition } from '@unlaxer/tramli';
import type { GenerationPlugin, PluginDescriptor } from '../api/types.js';
import type { DiagramBundle } from './types.js';
import { DiagramPlugin } from './diagram-plugin.js';

export class DiagramGenerationPlugin<S extends string> implements GenerationPlugin<FlowDefinition<S>, DiagramBundle> {
  private readonly delegate = new DiagramPlugin();

  descriptor(): PluginDescriptor {
    return {
      id: 'diagram',
      displayName: 'Diagram Generator',
      description: 'Generates Mermaid and data-flow bundles from a flow definition.',
    };
  }
  kind() { return 'GENERATION' as const; }

  generate(input: FlowDefinition<S>): DiagramBundle {
    return this.delegate.generate(input);
  }
}
